### PR TITLE
PODS-7556

### PR DESCRIPTION
**Pull request by emrawsy**

### CODE REVIEW

File: app/forms/SchemeDetails/SchemeDetailsFormProvider.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import forms.mappings.Mappings
+import play.api.data.Form
+
+class SchemeDetailsFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[Option[String]] =
+    Form(
+      "value" -> optionalText()
+        .verifying(maxLength(150, "schemeDetails.error.length"))
```
Comment (cyeniceri) [inline app/forms/SchemeDetails/SchemeDetailsFormProvider.scala:29]: Are we sure that there is a max length constraint here ?
It should be, but I couldn't verify that limit is 150 chars.

**Review by cyeniceri — CHANGES_REQUESTED** [review]

File: test/controllers/event1/member/SchemeDetailsControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        val result = route(application, request).value
+        val updatedAnswers = emptyUserAnswers.set(SchemeDetailsPage, "").success.value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual SchemeDetailsPage.navigate(waypoints, emptyUserAnswers, updatedAnswers).url
+        verify(mockUserAnswersCacheConnector, times(1)).save(any(), any(), any())(any(), any())
+      }
+    }
+
+
```
Comment (cyeniceri) [inline test/controllers/event1/member/SchemeDetailsControllerSpec.scala:138]: if there is maxlength validation, it's worth to have a bad request test here.

File: app/forms/SchemeDetails/SchemeDetailsFormProvider.scala (hunk 1 above)
Comment (emrawsy) [reply app/forms/SchemeDetails/SchemeDetailsFormProvider.scala:29]: @cyeniceri So the ticket says there shouldn't be any validation on this page however this doesn't seem right. Can remove the validation if needed however surely at the least a maxLength should be enforced?

File: app/forms/SchemeDetails/SchemeDetailsFormProvider.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import javax.inject.Inject
+
+import forms.mappings.Mappings
+import play.api.data.Form
+
+class SchemeDetailsFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[Option[String]] =
+    Form(
+      "value" -> optionalText()
```
Comment (cyeniceri) [inline app/forms/SchemeDetails/SchemeDetailsFormProvider.scala:28]: the fields name you used in SchemeDetails view :
schemeName and reference

so this provider is not validating anything.

and test is also not testing correct thing.

**Review by cyeniceri — CHANGES_REQUESTED** [review]

File: app/forms/SchemeDetails/SchemeDetailsFormProvider.scala (hunk 1 above)
Comment (cyeniceri) [reply app/forms/SchemeDetails/SchemeDetailsFormProvider.scala:29]: @emrawsy I agree with you, there should be a limit for both fields, we shouldn't assume it. 
Can you ask this to Sanjay and update the ticket as well for both fields ?

Comment (emrawsy) [reply app/forms/SchemeDetails/SchemeDetailsFormProvider.scala:29]: @cyeniceri sure, will do

**Pull request merged**